Sources/Commands/SwiftPackageTool.swift

```diff
@@ -367,60 +193,150 @@ public class SwiftPackageTool: SwiftTool<PackageMode, PackageToolOptions> {
+
+    override init(_ parser: ArgumentParser, args: [String]) throws {
+        // FIXME: Move these to subtool model.
+        modeArgument = parser.add(
+            positional: "mode", kind: PackageMode.self, usage: "(init|fetch|update|generate-xcodeproj|show-dependencies|dump-package)")
+
+        // FIXME: Add parser chaining and chain these options to edit subtool.
+        packageNameOption = parser.add(
+            option: "--name", kind: String.self, usage: "The yolo")
+        revisionOption = parser.add(
+            option: "--revision", kind: String.self, usage: "The yolo")
+        branchOption = parser.add(
+            option: "--branch", shortName: "-b", kind: String.self)
+        forceOption = parser.add(
+            option: "--force", shortName: "-f", kind: Bool.self)
+
+        // FIXME: These belongs to generate-xcode project subtool.
+        enableCodeCoverageOption = parser.add(
+            option: "--enable-code-coverage", kind: Bool.self)
+        xcconfigOverrides = parser.add(
+            option: "--xcconfig-overrides", kind: String.self)
+        outputOption = parser.add(option: "--output", kind: String.self, usage: "The output path for Xcode project")
+
+        // FIXME: This belongs to init subtool.
+        initTypeOption = parser.add(
+            option: "--type", kind: InitMode.self,
+            usage: "The init type: (empty|library|executable|system-module)")
+
+        // FIXME: Belongs to dump-package subtool.
+        inputOption = parser.add(option: "--input", kind: String.self, usage: "The path to Package.swift for dump-package command")
+
+        // FIXME: Belongs to show-dependencies subtool.
+        showDepsTypeOption = parser.add(option: "--format", kind: ShowDependenciesMode.self, usage: "format show-dependencies in: (text|dot|json)")
+
+        try super.init(parser, args: args)
     }
 }
 
-public func ==(lhs: PackageMode, rhs: PackageMode) -> Bool {
-    return lhs.description == rhs.description
+extension InitMode: StringEnumArgument {}
+extension ShowDependenciesMode: StringEnumArgument {}
+
+public enum PackageMode: StringEnumArgument {
+    case dumpPackage
+    case edit
+    case unedit
+    case fetch
+    case generateXcodeproj
+    case initPackage
+    case showDependencies
+    case reset
+    case resolve
+    case update
+    case version
+
+    public init?(rawValue: String) {
+        switch rawValue {
+        case "dump-package":
+            self = .dumpPackage
+        case "edit":
+            self = .edit
+        case "unedit":
+            self = .unedit
+        case "fetch":
+            self = .fetch
+        case "generate-xcodeproj":
+            self = .generateXcodeproj
+        case "init":
+            self = .initPackage
+        case "reset":
+            self = .reset
+        case "resolve":
+            self = .resolve
+        case "show-dependencies":
+            self = .showDependencies
+        case "update":
+            self = .update
```

Comment:
Could we make this cleaner by making this enum have a String raw value, and using those directly?
```
case initPackage = "init-package"
```
and so on? Would ditch a lot of boilerplate.